### PR TITLE
8289984: Files:isDirectory and isRegularFile methods not throwing SecurityException

### PR DESCRIPTION
**Pull request by LanceAndersen**
Hi all,

Please review this change to address the fact that we were missing a permissions check when the SecurityManager is enabled

The failing jck tests are passing locally and just waiting on another platform to complete on Mach5  and will be good to go

Best
Lance

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289984](https://bugs.openjdk.org/browse/JDK-8289984): Files:isDirectory and isRegularFile methods not throwing SecurityException

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to [b2d2cf75](https://git.openjdk.org/jdk/pull/9433/files/b2d2cf75c9d5d9cfc51ee7d44f7ea08d8734bd63)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to [b2d2cf75](https://git.openjdk.org/jdk/pull/9433/files/b2d2cf75c9d5d9cfc51ee7d44f7ea08d8734bd63)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9433/head:pull/9433` \
`$ git checkout pull/9433`

Update a local copy of the PR: \
`$ git checkout pull/9433` \
`$ git pull https://git.openjdk.org/jdk pull/9433/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9433`

View PR using the GUI difftool: \
`$ git pr show -t 9433`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9433.diff">https://git.openjdk.org/jdk/pull/9433.diff</a>

</details>

### CODE REVIEW

**Comment by LanceAndersen** [comment]
The previously failing jck tests are now passing on mach5

File: test/jdk/java/nio/file/Files/CheckPermissions.java [hunk 1]
```diff
@@ -219,6 +219,14 @@ public static void main(String[] args) throws IOException {
             exists(file);
```
Comment (irisclark) [inline test/jdk/java/nio/file/Files/CheckPermissions.java:219]: Sorry for the poor comment placement.  Do you need to add the bugid back at line 25?

**Review by irisclark — APPROVED** [review]

**Review by AlanBateman — APPROVED** [review]
Can you update the @bug tag before you integrate?

**Comment by LanceAndersen** [comment]
> Can you update the @bug tag before you integrate?

Done and pushed.  Thanks for the reminder

File: test/jdk/java/nio/file/Files/CheckPermissions.java (hunk 1 above)
Comment (LanceAndersen) [reply test/jdk/java/nio/file/Files/CheckPermissions.java:219]: Thanks for the reminder and all set.

**Pull request closed without merging**